iocage/lib/Jails.py

```diff
@@ -76,8 +77,32 @@ def _create_resource_instance(
         kwargs["host"] = self.host
         kwargs["zfs"] = self.zfs
         jail = self._class_jail(*args, **kwargs)
+
+        if jail.identifier in self.states:
+            self.logger.spam(
+                f"Injecting pre-loaded state to '{jail.humanreadable_name}'"
+            )
+            jail.jail_state = self.states[jail.identifier]
+
         return jail
 
+    def __iter__(
+        self
+    ) -> typing.Generator['iocage.lib.Resource.Resource', None, None]:
```

Comment:
is this necessary to be a `'ForwardReference'`?

Reply:
No, I don't see the requirement, but passed it as string for convenience with other situations where it makes a difference. (Idea for the future: We could aim to refactor all circular dependencies to a linear dependency tree.)